provider/gateway/rest/router.go

```diff
@@ -178,6 +183,86 @@ func newJwtServerRouter(addr sdk.Address, privateKey interface{}, jwtExpiresAfte
 	return router
 }
 
+func newResourceServerRouter(log log.Logger, providerAddr sdk.Address, publicKey *ecdsa.PublicKey, lokiGwAddr string) *mux.Router {
+	router := mux.NewRouter()
+
+	// add a middleware to verify the JWT provided in Authorization header
+	router.Use(func(next http.Handler) http.Handler {
```

Comment:
Can we make this middleware its own function in its own file somewhere? It's easier to read, test & debug that way

Reply:
Done